src/components/Organisms/MarketingPreferencesDS/_MarketingPrefsConfig.js

```diff
@@ -0,0 +1,144 @@
+import * as yup from 'yup';
+import 'yup-phone';
+import { merge } from 'lodash';
+
+const setInitialValues = overrideValues => {
+  const defaultValues = {
+    mp_email: '',
+    mp_mobile: '',
+    mp_phone: '',
+    mp_address1: '',
+    mp_address2: '',
+    mp_address3: '',
+    mp_town: '',
+    mp_country: '',
+    mp_postcode: '',
+    mp_permissionEmail: [],
+    mp_permissionPost: [],
+    mp_permissionPhone: [],
+    mp_permissionSMS: []
+  };
+
+  // Override with any custom options supplied
+  const updatedValues = merge(defaultValues, overrideValues);
+
+  return updatedValues;
+};
+
+const buildValidationSchema = overrideOptions => {
+  const defaultOptions = {
+    mp_permissionEmail: {
+      /**
+        * As per react-hook-form's validation requirements, sets the fields' required attribute
+        * for each checkbox option (Yes & No), a non-required field isn't rendered or included
+        * in the validation. As ERP *currently* needs values to formalise a user's opt-out,
+        * we'll set all fields to 'required' by default (once the option is chosen).
+        */
+      yes: true,
+      no: true,
+      // Allows complete removal of the option (checkboxes & fields) from both render & validation.
+      disableOption: false
+    },
+    mp_permissionSMS: {
+      yes: true,
+      no: true,
+      disableOption: false
+    },
+    mp_permissionPhone: {
+      yes: true,
+      no: true,
+      disableOption: false
+    },
+    mp_permissionPost: {
+      yes: true,
+      no: true,
+      disableOption: false
+    }
+  };
+
+  // Override with any custom options supplied
+  const mpValidationOptions = merge(defaultOptions, overrideOptions);
+
+  const mpValidationSchema = yup.object({
+    mp_email: yup.string().when('mp_permissionEmail', {
+      is: () => (!(mpValidationOptions.mp_permissionEmail.disableOption
+        || mpValidationOptions.mp_permissionEmail.hideInput)),
+      then: yup.string().email('Please enter a valid email address')
+      // Uses the config options to determine if 'required' validation is needed
+        .when('mp_permissionEmail', {
+          is: val => (mpValidationOptions.mp_permissionEmail[val]),
+          then: yup.string().required('Please enter your email address')
+        })
+    }),
+
+    mp_mobile: yup.string()
+      .when('mp_permissionSMS', {
+        is: val => (mpValidationOptions.mp_permissionSMS[val]),
+        then: yup.string().required('Please enter your mobile number')
+      })
+      .when('mp_permissionSMS', {
+        is: val => (!(mpValidationOptions.mp_permissionSMS.disableOption)
+        && mpValidationOptions.mp_permissionSMS[val]),
+        then: yup.string().phone('GB', false, 'Please enter a valid mobile number')
```

Comment:
Could we just add `.required('Please enter your mobile number')` here and have only one `when` condition?

Reply:
Ah yeah, nice; I just removed an additional check that Formik needed, but now there's a bit of duplication; will simplify where I can